pulp_container/app/registry_api.py

```diff
@@ -1009,3 +1049,61 @@ def get_response_data(signatures):
             }
             data.append(signature)
         return {"signatures": data}
+
+    def put(self, request, path, pk):
+        """Create a new signature from the received data."""
+        _, repository = self.get_dr_push(request, path)
+
+        try:
+            manifest = models.Manifest.objects.get(digest=pk)
+        except models.Manifest.DoesNotExist:
+            raise ManifestNotFound(reference=pk)
+
+        signature_payload = request.META["wsgi.input"].read(SIGNATURE_PAYLOAD_MAX_SIZE)
+        signature_raw = base64.b64decode(json.loads(signature_payload)["content"])
+        signature_json = extract_data_from_signature(signature_raw, manifest.digest)
```

Comment:
this utility accounts only for one signature extraction, PUT can contain 1+ signatures.

Reply:
Could you please provide a reference for this statement? I was looking into https://github.com/openshift/origin/pull/12504/files#diff-a215f9ef5344e2e492dbb2e3b82acf1d51f330e76e2c1dcc51b10a896e2d22fcR76 and https://docs.openshift.com/container-platform/3.11/admin_guide/image_signatures.html#writing-image-signatures-using-registry-api.

Reply:
I've been looking here https://github.com/mtrmac/image/commit/6c17ca34793b19accc3d278fc93ce68e9943fcb4#diff-6c682511ebec73d3ee49b3ed3d33708b9c598f9dd59c4f0685e793cc16baed8fR376
I do see discrepancies in different sources, yes.  let's discuss more on monday

Reply:
apparently the client parses the list of signatures and then sends N number of PUT requests  with one signature https://github.com/mtrmac/image/commit/6c17ca34793b19accc3d278fc93ce68e9943fcb4#diff-6c682511ebec73d3ee49b3ed3d33708b9c598f9dd59c4f0685e793cc16baed8fR408

Reply:
Exactly. So, I will not update the code to handle this situation.